### DISCUSSION TITLE
How to get only specific fields?

### DISCUSSION THREAD

**Discussion by heindrickdumdum0217** [wallneradam/esorm#17] [Q&A] [answered]
Is it possible to get only necessary fields?

**Comment by wallneradam** [comment, marked as answer]
@heindrickdumdum0217 

It is indeed possible:
```python
async def test():
    import esorm    
    class Model(esorm.ESModel):
        f_int: int = 0
        f_str: str = 'a'
    
    await esorm.setup_mappings()
    
    doc = Model(f_int=1, f_str='b')
    doc_id = await doc.save()
    assert doc_id is not None
    
    doc = await Model.get(doc_id)
    assert doc.f_str == 'b'
    assert doc.f_int == 1
    
    doc = await Model.search_one_by_fields(dict(_id=doc_id), _source=['f_str'])
    assert doc.f_str == 'b'
    assert doc.f_int == 0
    
    doc = await Model.search_one_by_fields(dict(_id=doc_id), _source=['f_int'])
    assert doc.f_str == 'a'
    assert doc.f_int == 1
```

Though `get` method has no `_source` parameter  (it will in the next release). So in this example I use the `search_one_by_fields` instead. 

Esorm search methods accepts any parameters `AsyncElasticsearch` can have, which most of the time in sync with ElasticSearch API. So `_source` can be used in queries. **Though you have to have defaults for the fields you don't want to query all the time.**

**Reply by heindrickdumdum0217** [reply]
The reason I'm going to set only specific fields is list api response can be bigger.
I don't think `search_one_by_field` meet my requirement.

**Reply by wallneradam** [reply]
What is your requirement? This can be used in any search method of Esorm, because it is the argument of underlying ElasticSearch ibrary.
